addons/cargo/functions/fnc_removeCargoItem.sqf

```diff
@@ -17,32 +17,33 @@
  */
 #include "script_component.hpp"
 
-params ["_itemClass", "_vehicle", ["_amount", 1]];
-TRACE_3("params",_itemClass,_vehicle,_amount);
+params ["_item", "_vehicle", ["_amount", 1]];
+TRACE_3("params",_item,_vehicle,_amount);
 
 private _loaded = _vehicle getVariable [QGVAR(loaded), []];
 
-private _itemsToRemove = [];
+private _itemsRemoved = 0;
 private _addedSpace = 0;
 
 {
-    if (count _itemsToRemove == _amount) exitWith {};
+    if (_itemsRemoved == _amount) exitWith {};
 
-    if (_x == _itemClass || {typeOf _x == _itemClass}) then {
-        _itemsToRemove pushBack _forEachIndex;
+    if (_x isEqualTo _item || {_x isEqualType objNull && {typeOf _x == _item}}) then {
```

Comment:
Would this not fail in the case where `_x` is an object, but not the same object as `_item`?